src/lib/db/feature-toggle-store.ts

```diff
@@ -397,29 +379,53 @@ export default class FeatureToggleStore implements IFeatureToggleStore {
         return toggle;
     }
 
-    async markPotentiallyStaleFeatures(
+    async updatePotentiallyStaleFeatures(
         currentTime?: string,
-    ): Promise<string[]> {
-        const query = this.db(TABLE)
-            .update('potentially_stale', true)
-            .whereRaw(
-                `? > (features.created_at + ((
-                    SELECT feature_types.lifetime_days
-                    FROM feature_types
-                    WHERE feature_types.id = features.type
-                ) * INTERVAL '1 day'))`,
-                [currentTime || this.db.fn.now()],
+    ): Promise<{ name: string; potentiallyStale: boolean }[]> {
+        const previousState = (
+            await this.db(TABLE)
+                .select(['name', 'potentially_stale'])
+                .whereNot('stale', true)
+        ).reduce((previousValue, current) => {
+            previousValue[current.name] = current.potentially_stale;
+            return previousValue;
+        }, {});
+
+        const updateQuery = this.db(TABLE)
```

Comment:
I have a feeling this will update all features every minute. I think we want to write only to database when the flag should change.

Reply:
Yeah, agree! I'll see if I can make knex do that. That should also alleviate the need for the manual diff.

Reply:
Maybe the previousState can already query the `state` the feature has to be after update and previousState.
Then on the next query, you can just send the filtered data.

Reply:
That's also a good idea! I thought I'd put it into a single query instead, something like this:

```sql
WITH
  feature_types_data AS (
    SELECT id, lifetime_days
    FROM feature_types),
  current_ts AS (
    SELECT CURRENT_TIMESTAMP AS current_ts),
  comparison AS (
    SELECT
	features.name,
	(current_ts.current_ts > (features.created_at + (feature_types_data.lifetime_days * INTERVAL '1 day'))) AS is_stale
    FROM
	features
    JOIN
	feature_types_data ON features.type = feature_types_data.id
    CROSS JOIN
	current_ts)
UPDATE features
SET potentially_stale = comparison.is_stale
FROM comparison
WHERE features.name = comparison.name
  AND (
    (comparison.is_stale AND NOT potentially_stale = TRUE)
    OR (NOT comparison.is_stale AND potentially_stale = TRUE)
  )
  AND NOT stale = TRUE
RETURNING features.name, potentially_stale;
```

But I can explore both options 💁🏼

Reply:
Solution discussed with @sjaanus in Slack was implemented in 66e838d642d363cc06a25ae3adcb9cee4820b57d